### PR TITLE
[ci] add shallow_since in rules_perl

### PR DESCRIPTION
**Pull request by aslonnie**
so that it does not need to clone the entire repo.

this is recommended by latest bazel build command.

### CODE REVIEW

File: bazel/ray_deps_setup.bzl [hunk 1]
```diff
@@ -261,6 +261,7 @@ def ray_deps_setup():
         name = "rules_perl",
         remote = "https://github.com/bazelbuild/rules_perl.git",
         commit = "022b8daf2bb4836ac7a50e4a1d8ea056a3e1e403",
+        shallow_since = "1663780239 -0700",
```
Comment (can-anyscale) [inline bazel/ray_deps_setup.bzl:264]: maybe put a link to bazel documentation?

**Review by can-anyscale — APPROVED** [review]

File: bazel/ray_deps_setup.bzl (hunk 1 above)
Comment (aslonnie) [reply bazel/ray_deps_setup.bzl:264]: it's fairly standard thing for using `git_repository`: https://bazel.build/rules/lib/repo/git

the hint to add a `shallow_since` is provided by bazel build process in the build logs.

Comment (aslonnie) [reply bazel/ray_deps_setup.bzl:266]: added comments.

**Pull request merged**